### PR TITLE
41 display highlight area on the minimap

### LINKED ISSUE #41: display highlight area on the minimap (minimap orientation)
**As a** Developer, **I want** an adjustable highlight area on the minimap, **so that** I can point out certain parts of the minimap.

### Acceptance Criteria

Constraints:
- [x] There is a new layer drawn on top of the minimap, that allows other graphics to be displayed;
- [x] The area is highlighted via the graphic of a white rectangle outline;
- [x] The highlight area is adjustable in position and size;
- [x] The size of the area doesn't get bigger then the minimap (cap width & height).

### Additional Resources:

This is what the highlight area might look like:
329233649-a6133e05-872e-466f-be8f-0f9d341ed983
grafik

 (Ignore the rest of the buttons/features of this minimap)

### PR DESCRIPTION
**Pull request by RukiyyeE**
resolves #41 

## Description
This ticket is the second part of the previous issue 12. This PR implements the feature to add a highlight area on the minimap, which can be adjusted in size and position based on normalized coordinates calculated in Ticket #65.

## Changes

The following changes were made in **cellpose/gui/guiparts.py**:

- Added `set_highlight_area` method to the `MinimapWindow class`:
   - Calculates the position and size of the highlight area based on normalized coordinates.
   - Caps the width and height of the highlight area to ensure it does not exceed the minimap size.
   - Sets the position and size of the highlight area on the minimap.
   
- Integrated the highlight area into the minimap as a `pg.RectROI` object:
   - Created and added a rectangle highlight area to the minimap.
   - Set the highlight area with an initial position of `[0, 0]` and size `[100, 100]`, outlined in white with a 3-pixel width.

## Testing

To test the highlight area, you can change the normalized values for the coordinates and dimensions in the method call self.set_highlight_area(normalized_x, normalized_y, normalized_width, normalized_height) within the MinimapWindow(QDialog) class to draw a rectangle at the corresponding position and size.

### CODE REVIEW

File: cellpose/gui/guiparts.py
Comment (mariehartung) [inline cellpose/gui/guiparts.py:456]: I would be good to change this size to the size of the minimap if possible (see picture in the main review)

File: cellpose/gui/guiparts.py
Comment (mariehartung) [inline cellpose/gui/guiparts.py]: the method is called onViewChanged ;)

File: cellpose/gui/guiparts.py
Comment (mariehartung) [inline cellpose/gui/guiparts.py]: Do we need these returns?

**Review by mariehartung — CHANGES_REQUESTED** [review]
### Main Review

Good job! Good implementation overall! 

#### Change Requests
Right now, the coordinates seem to be not quite correct:
e.g. for the values (0.2, 0.1, 0.5, 0.5) it looks like this:
image
or for (1, 1, 0.3, 0.5):
image
Width and height seem to be fine, but the coordinates don't quite correspond I believe.
Edit: I am not quite sure but I think there might be some kind of "border" for the coordinates

If the values for width/height are bigger than one, only two borders can be seen:
image
This is only a small issue, but it would be great if you could find a way to fix it :)

I'd also recommend using the full size of the minimap as a starting size if possible for that is more accurate to the first position of the image (right now it looks like this at the beginning):
image

#### Further comments

I also get this error message: `An error occurred while changing the view: 'NoneType' object has no attribute 'shape'`, but maybe that will be solved once this ticket and former ticket 40 are combined.

I also don't see yet what we need the returns for.

**Review by NicolasDelgadoL — DISMISSED** [review]
Very nice implementation! With the latest changes, the rectangle exactly shows the view area on the main window. The highlight area's maximal values match the size of the minimap. The parameters of your method match the values computed in ticket #40, which pave the way for a smooth solution to ticket #42.
I went ahead and deleted two lines that that imported the PIL package and were causing some trouble on my computer. Other than that, nothing to add :)

**Comment by RukiyyeE** [comment]
Thank you for your review. I have fixed the issues. The only remaining issue might be the error message: `An error occurred while changing the view: 'NoneType' object has no attribute 'shape'`. This started to occur after the merge commit d79395e.

File: cellpose/gui/guiparts.py
Comment (mariehartung) [inline cellpose/gui/guiparts.py]: This already fixed most of the issues, but I think it might still be causing a small issue (see review)

**Review by mariehartung — CHANGES_REQUESTED** [review]
## Second review

Thank you for implementing the change requests so quickly!
Almost everything works perfectly, also in combination with the normalization thanks to #71 

### Small Issues
#### Change request
Almost everything works perfectly, but there still is one problem:
If I move the image out of the view box, the box only adapts to this if it's on the right side or at the bottom. If it's on the left/at the top, it looks e.g. like this:
image
I have marked the lines I suspect are causing this in your code.

#### Information about the error (not a change request)
I also still get the error message as well. I think it might be caused partially by my code in the previous #65, for this code alone raises this error when zooming in an empty view box:
`An error occurred while changing the view: 'NoneType' object has no attribute 'shape'`
This looks quite identical to me, so I presume that either the combination of our tickets, or just the empty view box itself raises this error; Therefore I would separate this issue from your ticket :)
Edit: I am quite sure it is caused by the exception in former #65 in gui.py from line 482 on

**Review by mariehartung — APPROVED** [review]
### Approval

Thank you for the changes! I will approve now.

**Review by NicolasDelgadoL — APPROVED** [review]

**Pull request merged**
